src/System.CommandLine/Invocation/InvocationContext.cs

```diff
@@ -73,39 +81,40 @@ public ParseResult ParseResult
         /// <remarks>As the <see cref="InvocationContext"/> is passed through the invocation pipeline to the <see cref="ICommandHandler"/> associated with the invoked command, only the last value of this property will be the one applied.</remarks>
         public IInvocationResult? InvocationResult { get; set; }
 
-        internal event Action<CancellationTokenSource> CancellationHandlingAdded
-        {
-            add
-            {
-                if (_cts is not null)
-                {
-                    throw new InvalidOperationException("Handlers must be added before adding cancellation handling.");
-                }
-
-                _cancellationHandlingAddedEvent += value;
-            }
-            remove => _cancellationHandlingAddedEvent -= value;
-        }
-
         /// <summary>
-        /// Gets token to implement cancellation handling.
+        /// Gets a cancellation token that can be used to check if cancellation has been requested.
         /// </summary>
-        /// <returns>Token used by the caller to implement cancellation handling.</returns>
-        public CancellationToken GetCancellationToken()
+        public CancellationToken CancellationToken => _lazyCancellationToken.Value;
+
+        private CancellationToken BuildCancellationToken()
         {
-            if (_cts is null)
+            if (_linkedTokens is not null)
             {
-                _cts = new CancellationTokenSource();
-                _cancellationHandlingAddedEvent?.Invoke(_cts);
+                var tokens = new List<CancellationToken>(_linkedTokens.Count + 1);
+                tokens.Add(_cancellationToken);
+                tokens.AddRange(_linkedTokens);
+                _linkedTokensSource = CancellationTokenSource.CreateLinkedTokenSource(tokens.ToArray());
+                return _linkedTokensSource.Token;
             }
-
-            return _cts.Token;
+            return _cancellationToken;
         }
 
+
         /// <inheritdoc />
         public void Dispose()
         {
+            _linkedTokensSource?.Dispose();
+            _linkedTokensSource = null;
             (Console as IDisposable)?.Dispose();
         }
+
+        public void AddLinkedCancellationToken(CancellationToken token)
+        {
+            if (_lazyCancellationToken.IsValueCreated)
+            {
+                throw new InvalidOperationException($"Cannot add additional linked cancellation tokens once {nameof(InvocationContext)}.{nameof(CancellationToken)} has been invoked");
```

Comment:
```suggestion
                throw new InvalidOperationException($"Cannot add additional linked cancellation tokens once {nameof(InvocationContext)}.{nameof(CancellationToken)} has been cancelled");
```

Reply:
I don't think this is correct. The failure occurs if the Lazy has already been evaluated, meaning something requested the `CancellationToken` from the context. It may or may not be cancelled at this point.

Reply:
This initially caught my eye  because I wasn't sure what it means to "invoke" a `CancellationToken`. If I saw this exception message, I'm not sure I'd understand what I should do differently. What can we do to make this error clearly actionable for developers?